WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormActionsFactory.swift

```diff
@@ -426,7 +424,7 @@ private extension ProductFormActionsFactory {
             return (product.upsellIDs.count > 0 || product.crossSellIDs.count > 0)
         // Downloadable files. Only core product types for downloadable files are able to handle downloadable files.
         case .downloadableFiles:
-            return product.downloadable
+            return product.downloadable && product.downloadableFiles.isNotEmpty
```

Comment:
We should keep the row when the product is downloadable, but no file has been uploaded yet for better visibility of the section - WDYT?

Reply:
I'm not sure, I updated it to this for two reasons:
1. It aligns with what we do in Android.
2. It follows what we do with the other rows (like categories and tags)

But I don't have a strong opinion, let me know if knowing the above two reasons you still think we should revert it.

Reply:
I'm enabling auto-merge on this now, but feel free to share your thoughts later, and we can always revert the change on a second PR if needed.

Reply:
I think files should be essential to downloadable products, so just the check for downloadable should be sufficient to show the row for uploading files.

Reply:
OK, done in a6c9677e5fc9c6f9857c7b4284d50831a4979af8